hoodie-client/src/test/java/com/uber/hoodie/io/strategy/TestHoodieCommitFile.java

```diff
@@ -0,0 +1,151 @@
+import com.uber.hoodie.common.model.HoodieCommitMetadata;
+import com.uber.hoodie.common.model.HoodieRecord;
+import com.uber.hoodie.common.model.HoodieTestUtils;
+import com.uber.hoodie.common.table.HoodieTableMetaClient;
+import com.uber.hoodie.common.table.HoodieTimeline;
+import com.uber.hoodie.common.table.timeline.HoodieInstant;
+import com.uber.hoodie.common.util.FSUtils;
+import com.uber.hoodie.config.HoodieCompactionConfig;
+import com.uber.hoodie.config.HoodieIndexConfig;
+import com.uber.hoodie.config.HoodieStorageConfig;
+import com.uber.hoodie.config.HoodieWriteConfig;
+import com.uber.hoodie.index.HoodieIndex;
+import com.uber.hoodie.table.HoodieTable;
+import org.apache.commons.io.IOUtils;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.SQLContext;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.*;
+import java.util.*;
+
+import static org.junit.Assert.assertTrue;
+
+public class TestHoodieCommitFile {
+  private transient JavaSparkContext jsc = null;
+  private transient SQLContext sqlContext;
+  private String basePath = null;
+  private String partitionPath = null;
+  private FileSystem fs = null;
+  private transient HoodieTestDataGenerator
+          dataGen = null;
+  private HoodieTableMetaClient metaClient;
+
+
+  @Before
+  public void init() throws IOException {
+    // Initialize a local spark env
+    SparkConf sparkConf = new SparkConf().setAppName("TestHoodieClient").setMaster("local[4]");
+    jsc = new JavaSparkContext(HoodieReadClient.addHoodieSupport(sparkConf));
+
+    //SQLContext stuff
+    sqlContext = new SQLContext(jsc);
+
+    // Create a temp folder as the base path
+    TemporaryFolder folder = new TemporaryFolder();
+    folder.create();
+    basePath = folder.getRoot().getAbsolutePath();
+    String[] partitionPath = {"2016/05/19"};
+    HoodieTestUtils.init(basePath);
+    fs = FSUtils.getFs();
+
+
+    dataGen = new HoodieTestDataGenerator(partitionPath);
+
+    this.metaClient = HoodieTestUtils.initOnTemp();
+    this.basePath = metaClient.getBasePath();
+  }
+
+  private HoodieWriteConfig.Builder getConfigBuilder() {
+    return HoodieWriteConfig.newBuilder().withPath(basePath)
+            .withSchema(HoodieTestDataGenerator.TRIP_EXAMPLE_SCHEMA).withParallelism(2, 2)
+            .withCompactionConfig(HoodieCompactionConfig.newBuilder().compactionSmallFileSize(1024 * 1024).build())
+            .withStorageConfig(HoodieStorageConfig.newBuilder().limitFileSize(1024 * 1024).build())
+            .forTable("test-trip-table").withIndexConfig(
+                    HoodieIndexConfig.newBuilder().withIndexType(HoodieIndex.IndexType.BLOOM).build());
+  }
+
+  @Test
+  public void testCommitWritesRelativePaths() throws Exception {
```

Comment:
Can this be a test in TestHoodieClient? Lets not introduce a new test class. TestHoodieClient checks for commits as well in other tests.